src/renderer/components/modules/AddAccountWindow.tsx

```diff
@@ -72,23 +90,41 @@ const AddAccountWindow = (props: AddAccountWindowProps) => {
       password: data.get("password") as string,
       imapServer: data.get("server") as string,
       pop3Server: data.get("server") as string,
+      openai_api_key: data.get("openai_api_key") as string,
     };
     console.log(req);
-    newEmailAccount(req)
-      .then((errMsg) => {
-        setLoading(false);
-        if (errMsg === "") {
-          props.callGetUserInfo();
-          props.setAddAccount(false);
-        } else {
-          console.log("error when adding mailbox:", errMsg);
-          setErrorMsg(errMsg);
-        }
-      })
-      .catch((err) => {
-        console.log(err);
-      });
-  };
+    // todo: add openai api key to local storage
+    updateSettings({ apiKey: req.openai_api_key })
+    .then((errMsg) => {
+      setLoading(false);
```

Comment:
Duplicated setLoading. We should only set loading to false after there is an error or the whole operation is complete.

Reply:
<img width="288" alt="image" src="https://github.com/dongmingli-Ben/email-wizard-desktop/assets/65433575/d7075e01-bdfe-4a66-a2e9-ccd45d4c491b">
A quick question, here the setLoading is set to false after executing no matter whether newEmailAccount is failed. Is it because the operation is ending here?